snakemake/benchmark.py

```diff
@@ -386,3 +388,63 @@ def write_benchmark_records(records, path):
     """Write benchmark records to file at path"""
     with open(path, "wt") as f:
         print_benchmark_records(records, f)
+
+
+def gather_benchmark_records(benchmark_jobs, persistence):
+    """
+    Gather benchmark from given files.
+
+    Args:
+        benchmark_jobs (List[Job]): A list of jobs to extract benchmark from
+    Return:
+        (DataFrame): A Pandas DataFrame object with jobid as index,
+            containing benchmark values for all jobs
+    """
+    benchmarks = pd.DataFrame()
+    for job in benchmark_jobs:
+        if not job._benchmark.exists:
+            raise IOFileException(
+                "Error: \n"
+                + f"Benchmark file {job._benchmark} does not "
+                + f"exist for rule {job.rule.name} \n"
+                + "A complete run is required to print all benchmark metrics."
+            )
+        wildcard_str = ";".join(
+            ["NA"]
+            if len(job.wildcards_dict) == 0
+            else [f"{name}={value}" for name, value in job.wildcards_dict.items()]
```

Comment:
This simple encoding will not be parseable if any of name/value contain either of `;` or `=`

One could quote encode in those cases and escape quotes with backslash.